unittest/TestUtil.hpp

```diff
@@ -20,6 +20,8 @@
 
 #include "system.hpp"
 
+#include <algorithm>
+#include <ostream>
```

Comment:
Suggestion: Move these to the .cpp file instead since they are not required by anything in the header file.